tests/utils_tests/test_lazyobject.py

```diff
@@ -19,6 +19,57 @@ def __eq__(self, other):
         return self.foo == other.foo
 
 
+class BaseBaz(object):
+    """
+    A base class with a funky __reduce__ method.
+
+    Meant to simulate the __reduce__ method of Model, which sets
```

Comment:
Is this better than testing with an actual model? I imagine something in model could change in the future that could cause this to regress if we don't add a test for that.